xarray/core/dataset.py

```diff
@@ -2667,7 +2667,8 @@ def _rename_indexes(self, name_dict, dims_set):
                     verify_integrity=False,
                 )
             else:
-                index = pd.Index(v, name=new_name)
+                index = v
+                index.name = new_name
```

Comment:
Alternatively there is a `index = v._shallow_copy(name=new_name)` method. It
> *[] is an internal non-public method*
in pandas. Should I use it anyway?

Reply:
While I think it's fairly stable, you're right—best practice is indeed to avoid. 

I think the current code will mutate the existing index name, is that correct? We should at least test to ensure that doesn't happen. I think `.rename` rather than direct assignment will avoid this (and calls `_shallow_copy` from within pandas so we're safe)

FWIW I think we can replace https://github.com/pydata/xarray/pull/3532/files#diff-921db548d18a549f6381818ed08298c9R2662-R2667 also with `.rename`, but it's minor and doesn't need to be in this PR

Reply:
Oh, there is even `rename` - I did not see that, will switch to it.